UnityProject/Assets/Scripts/Systems/Disposals/DisposalTraversal.cs

```diff
@@ -262,8 +299,10 @@ private void TryDamageTileFromEjection(Vector3Int localPosition)
 		/// </summary>
 		private void DespawnContainerAndFinish()
 		{
-			_ = Despawn.ServerSingle(virtualContainer.gameObject);
+			virtualContainer.SelfControlled = false;
 			TraversalFinished = true;
+			if (virtualContainer.SelfControlled) DisposalsManager.Instance.FinishDisposal(this);
+			_ = Despawn.ServerSingle(virtualContainer.gameObject);
```

Comment:
No point for the if condition, right?

Also, disposals manager is responsible for removing finished instances from its own list, so it would now be unnecessarily removing the same instance again.

Reply:
Alright, buyt SelfControlled will always be false at that point.

Reply:
oh, you're right.

Reply:
fixed